# Gemma-Mitra Embedding Protocol Comparison

This notebook compares four sentence-embedding protocols on the same two Tibetan passages:

- asymmetric `A -> B` retrieval mode (`query_corpus`)
- asymmetric `B -> A` retrieval mode (`query_corpus`, reversed)
- symmetric `query_query`
- symmetric `raw_raw`

The goal is to make the protocol choice inspectable before scaling to corpus-level analysis.

In [1]:
!pip install --no-build-isolation pyewts botok
!pip install -r /content/embedding-model-for-tibetan/requirements.txt


In [2]:
!ls -la

total 24
drwxr-xr-x 1 root root 4096 Apr 26 22:50 .
drwxr-xr-x 1 root root 4096 Apr 26 22:40 ..
drwxr-xr-x 3 root root 4096 Apr 26 22:50 .cache
drwxr-xr-x 1 root root 4096 Mar 30 13:34 .config
drwxr-xr-x 9 root root 4096 Apr 26 22:44 embedding-model-for-tibetan
drwxr-xr-x 1 root root 4096 Mar 30 13:34 sample_data


In [3]:
!nvidia-smi
import torch

print("cuda_available:", torch.cuda.is_available())
if torch.cuda.is_available():
    print("gpu_name:", torch.cuda.get_device_name(0))
    props = torch.cuda.get_device_properties(0)
    print("total_vram_gb:", round(props.total_memory / 1024**3, 2))


Sun Apr 26 23:02:26 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.82.07              Driver Version: 580.82.07      CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Tesla T4                       Off |   00000000:00:04.0 Off |                    0 |
| N/A   46C    P8             10W /   70W |       0MiB /  15360MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

In [4]:
!free -h

               total        used        free      shared  buff/cache   available
Mem:            12Gi       2.9Gi       2.6Gi       2.0Mi       7.2Gi       9.5Gi
Swap:             0B          0B          0B


In [5]:
!pip install -U bitsandbytes accelerate


In [6]:
import os
import subprocess
import sys
from pathlib import Path

AUTO_INSTALL_DEPS = True
REPO_URL = "https://github.com/ten-jampa/embedding-model-for-tibetan.git"
COLAB_REPO_DIR = Path("/content/embedding-model-for-tibetan")


def running_in_colab() -> bool:
    return "google.colab" in sys.modules


def find_project_root(start: Path) -> Path:
    current = start.resolve()
    for candidate in [current, *current.parents]:
        if (candidate / "tibetan_pipeline").exists() and (candidate / "README.md").exists():
            return candidate
    raise RuntimeError("Could not find project root containing tibetan_pipeline/ and README.md")


def ensure_repo() -> Path:
    env_root = os.environ.get("TIBETAN_PIPELINE_PROJECT_ROOT")
    if env_root:
        candidate = Path(env_root).expanduser().resolve()
        if candidate.exists():
            return find_project_root(candidate)

    if running_in_colab():
        if not COLAB_REPO_DIR.exists():
            subprocess.run(["git", "clone", REPO_URL, str(COLAB_REPO_DIR)], check=True)
        return find_project_root(COLAB_REPO_DIR)

    return find_project_root(Path.cwd())


def install_runtime_deps(project_root: Path) -> None:
    requirements_path = project_root / "requirements.txt"
    subprocess.run([sys.executable, "-m", "pip", "install", "--no-build-isolation", "pyewts", "botok"], check=True)
    subprocess.run([sys.executable, "-m", "pip", "install", "-r", str(requirements_path)], check=True)


PROJECT_ROOT = ensure_repo()
if AUTO_INSTALL_DEPS:
    install_runtime_deps(PROJECT_ROOT)

if str(PROJECT_ROOT) not in sys.path:
    sys.path.insert(0, str(PROJECT_ROOT))

print(f"PROJECT_ROOT={PROJECT_ROOT}")
print(f"PYTHON={sys.executable}")
print(f"IN_COLAB={running_in_colab()}")
print("Note: Colab clone uses origin/main unless you point TIBETAN_PIPELINE_PROJECT_ROOT at another checkout.")

from tibetan_pipeline import TibetanResearchSDK
from tibetan_pipeline.sdk import EmbeddingView


PROJECT_ROOT=/content/embedding-model-for-tibetan
PYTHON=/usr/bin/python3
IN_COLAB=True
Note: Colab clone uses origin/main unless you point TIBETAN_PIPELINE_PROJECT_ROOT at another checkout.


## Using 8bit model

In [7]:
import numpy as np
import pandas as pd

from tibetan_pipeline import TibetanResearchSDK

sdk = TibetanResearchSDK(
    engine="botok_ours",
    source_format="unicode",
    model_id="buddhist-nlp/gemma-2-mitra-e",
    device="cuda",
    batch_size=1,
    embedding_progress="batch",
    torch_dtype="float16",
    load_in_8bit=True,
)


In [8]:
print("device:", sdk.device)
print("torch_dtype:", sdk.torch_dtype)
print("load_in_8bit:", sdk.load_in_8bit)
print("batch_size:", sdk.batch_size)


device: cuda
torch_dtype: float16
load_in_8bit: True
batch_size: 1


In [9]:
from time import perf_counter

t0 = perf_counter()
test_emb = sdk.embed_sentences(["ང་བོད་པ་ཡིན་"], is_query=True)
t1 = perf_counter()
print("warmup_seconds:", round(t1 - t0, 2))
print("shape:", test_emb.embeddings.shape)


/usr/local/lib/python3.12/dist-packages/huggingface_hub/utils/_auth.py:104: UserWarning: 
Error while fetching `HF_TOKEN` secret value from your vault: 'Requesting secret HF_TOKEN timed out. Secrets can only be fetched when running from the Colab UI.'.
You are not authenticated with the Hugging Face Hub in this notebook.
If the error persists, please let us know by opening an issue on GitHub (https://github.com/huggingface/huggingface_hub/issues/new).
  warnings.warn(
`torch_dtype` is deprecated! Use `dtype` instead!
The following generation flags are not valid and may be ignored: ['cache_implementation']. Set `TRANSFORMERS_VERBOSITY=info` for more details.


Loading weights:   0%|          | 0/464 [00:00<?, ?it/s]

[embed] backend=gemma-last-token device=cuda total_sentences=1 batch_size=1 batches=1
[embed] batch 1/1 sentences 1-1/1
warmup_seconds: 152.88
shape: (1, 3584)


In [13]:
import time

t0 = time.perf_counter()
test_emb2 = sdk.embed_sentences(["ང་བོད་ལ་དགའ་"], is_query=False)
t1 = time.perf_counter()

print("second_call_seconds:", round(t1 - t0, 2))
print("shape:", test_emb2.embeddings.shape)


[embed] backend=gemma-last-token device=cuda total_sentences=1 batch_size=1 batches=1
[embed] batch 1/1 sentences 1-1/1
second_call_seconds: 0.79
shape: (1, 3584)


In [21]:
import torch
print(torch.cuda.memory_allocated() / 1024**3)
print(torch.cuda.memory_reserved() / 1024**3)


9.479180335998535
9.80859375


## Input Passages

Replace these with any two Tibetan paragraphs you want to study. Keep them as multi-sentence passages so the notebook can expose the sentence-level matrix behavior.

In [14]:
text_a = """
རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚོས་མཆོད་པ་བསྒྲིགས། ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས་སྒོམ་པ་བྱས། མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐོས་པ་བྱུང་།
""".strip()

text_b = """
བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་མཆོད་པ་བཤམས། ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛིན་བསྐྱངས། མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམས་ཞི་བར་གྱུར།
""".strip()

print(text_a)
print()
print(text_b)

རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚོས་མཆོད་པ་བསྒྲིགས། ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས་སྒོམ་པ་བྱས། མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐོས་པ་བྱུང་།

བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་མཆོད་པ་བཤམས། ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛིན་བསྐྱངས། མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམས་ཞི་བར་གྱུར།


In [15]:
seg_a = sdk.segment_text(text_a)
seg_b = sdk.segment_text(text_b)

segments_a = [segment for segment in seg_a.segments if segment.strip()]
segments_b = [segment for segment in seg_b.segments if segment.strip()]

segment_overview = pd.DataFrame(
    [
        {
            "text": "A",
            "segment_count": len(segments_a),
            "max_segment_chars": max((len(s) for s in segments_a), default=0),
            "mean_segment_chars": round(np.mean([len(s) for s in segments_a]), 2) if segments_a else 0,
        },
        {
            "text": "B",
            "segment_count": len(segments_b),
            "max_segment_chars": max((len(s) for s in segments_b), default=0),
            "mean_segment_chars": round(np.mean([len(s) for s in segments_b]), 2) if segments_b else 0,
        },
    ]
)

display(segment_overview)
display(pd.DataFrame({"A_segments": pd.Series(segments_a), "B_segments": pd.Series(segments_b)}))

,text,segment_count,max_segment_chars,mean_segment_chars
0,A,3,64,59.67
1,B,3,59,57.33


,A_segments,B_segments
0,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
1,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
2,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...


In [16]:
def pairwise_protocol(protocol_name: str, sentences_a: list[str], sentences_b: list[str]):
    if protocol_name == "query_corpus":
        emb_a = sdk.embed_sentences(sentences_a, is_query=True)
        emb_b = sdk.embed_sentences(sentences_b, is_query=False)
    elif protocol_name == "query_query":
        emb_a = sdk.embed_sentences(sentences_a, is_query=True)
        emb_b = sdk.embed_sentences(sentences_b, is_query=True)
    elif protocol_name == "raw_raw":
        emb_a = sdk.embed_sentences(sentences_a, is_query=False)
        emb_b = sdk.embed_sentences(sentences_b, is_query=False)
    else:
        raise ValueError(f"Unsupported protocol: {protocol_name}")

    return sdk.pairwise_from_embedding_views(emb_a, emb_b, top_k=min(10, len(sentences_a) * len(sentences_b)))


def summarize_view(label: str, view) -> dict[str, object]:
    matrix = view.similarity_matrix
    return {
        "label": label,
        "shape": tuple(matrix.shape),
        "max_score": round(float(np.max(matrix)), 4) if matrix.size else None,
        "mean_score": round(float(np.mean(matrix)), 4) if matrix.size else None,
        "p95_score": round(float(np.percentile(matrix, 95)), 4) if matrix.size else None,
        "mean_best_row": round(float(np.mean(np.max(matrix, axis=1))), 4) if matrix.size else None,
        "mean_best_col": round(float(np.mean(np.max(matrix, axis=0))), 4) if matrix.size else None,
    }


def matrix_frame(view, row_label: str, col_label: str) -> pd.DataFrame:
    return pd.DataFrame(
        view.similarity_matrix,
        index=[f"{row_label}{i:02d}" for i in range(len(view.segments_a))],
        columns=[f"{col_label}{j:02d}" for j in range(len(view.segments_b))],
    )


In [17]:
view_ab = pairwise_protocol("query_corpus", segments_a, segments_b)
view_ba = pairwise_protocol("query_corpus", segments_b, segments_a)
view_qq = pairwise_protocol("query_query", segments_a, segments_b)
view_rr = pairwise_protocol("raw_raw", segments_a, segments_b)

summary_df = pd.DataFrame(
    [
        summarize_view("A->B query_corpus", view_ab),
        summarize_view("B->A query_corpus", view_ba),
        summarize_view("A<->B query_query", view_qq),
        summarize_view("A<->B raw_raw", view_rr),
    ]
)

display(summary_df)

[embed] backend=gemma-last-token device=cuda total_sentences=3 batch_size=1 batches=3
[embed] batch 1/3 sentences 1-1/3
[embed] batch 2/3 sentences 2-2/3
[embed] batch 3/3 sentences 3-3/3
[embed] backend=gemma-last-token device=cuda total_sentences=3 batch_size=1 batches=3
[embed] batch 1/3 sentences 1-1/3
[embed] batch 2/3 sentences 2-2/3
[embed] batch 3/3 sentences 3-3/3
[embed] backend=gemma-last-token device=cuda total_sentences=3 batch_size=1 batches=3
[embed] batch 1/3 sentences 1-1/3
[embed] batch 2/3 sentences 2-2/3
[embed] batch 3/3 sentences 3-3/3
[embed] backend=gemma-last-token device=cuda total_sentences=3 batch_size=1 batches=3
[embed] batch 1/3 sentences 1-1/3
[embed] batch 2/3 sentences 2-2/3
[embed] batch 3/3 sentences 3-3/3
[embed] backend=gemma-last-token device=cuda total_sentences=3 batch_size=1 batches=3
[embed] batch 1/3 sentences 1-1/3
[embed] batch 2/3 sentences 2-2/3
[embed] batch 3/3 sentences 3-3/3
[embed] backend=gemma-last-token device=cuda total_sentences

,label,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A->B query_corpus,"(3, 3)",0.7586,0.4951,0.7467,0.7234,0.7234
1,B->A query_corpus,"(3, 3)",0.7636,0.5047,0.7577,0.7360,0.7360
2,A<->B query_query,"(3, 3)",0.8143,0.5430,0.8000,0.7771,0.7771
3,A<->B raw_raw,"(3, 3)",0.8078,0.5382,0.8060,0.7800,0.7800


## Similarity Matrices

Look for two things:

- whether `A -> B` and `B -> A` are materially different
- whether the symmetric protocols preserve the same strongest alignments

In [18]:
display(matrix_frame(view_ab, "A", "B").round(4))
display(matrix_frame(view_ba, "B", "A").round(4))
display(matrix_frame(view_qq, "A", "B").round(4))
display(matrix_frame(view_rr, "A", "B").round(4))

,B00,B01,B02
A00,0.7586,0.4143,0.3951
A01,0.4115,0.6828,0.3473
A02,0.3709,0.3467,0.7288


,A00,A01,A02
B00,0.7636,0.4471,0.4038
B01,0.4028,0.6957,0.3367
B02,0.3701,0.3734,0.7487


,B00,B01,B02
A00,0.8143,0.4601,0.4269
A01,0.4711,0.7384,0.3964
A02,0.4291,0.3724,0.7785


,B00,B01,B02
A00,0.8033,0.4325,0.4116
A01,0.4517,0.7288,0.3920
A02,0.4250,0.3907,0.8078


In [19]:
topk_tables = {
    "A->B query_corpus": view_ab.topk_dataframe(),
    "B->A query_corpus": view_ba.topk_dataframe(),
    "A<->B query_query": view_qq.topk_dataframe(),
    "A<->B raw_raw": view_rr.topk_dataframe(),
}

for label, table in topk_tables.items():
    print(f"## {label}")
    display(table.head(10))

## A->B query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.758572,0,0,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
1,2,0.728814,2,2,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
2,3,0.682754,1,1,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
3,4,0.414284,0,1,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
4,5,0.411482,1,0,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
5,6,0.395080,0,2,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
6,7,0.370906,2,0,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
7,8,0.347286,1,2,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
8,9,0.346737,2,1,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...


## B->A query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.763627,0,0,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...
1,2,0.748733,2,2,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...
2,3,0.695712,1,1,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...
3,4,0.447115,0,1,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...
4,5,0.403808,0,2,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...
5,6,0.402819,1,0,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...
6,7,0.373356,2,1,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...
7,8,0.370057,2,0,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...
8,9,0.336742,1,2,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...


## A<->B query_query


,rank,score,i,j,sentence_a,sentence_b
0,1,0.814318,0,0,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
1,2,0.778494,2,2,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
2,3,0.738439,1,1,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
3,4,0.471087,1,0,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
4,5,0.460115,0,1,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
5,6,0.429080,2,0,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
6,7,0.426863,0,2,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
7,8,0.396416,1,2,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
8,9,0.372376,2,1,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...


## A<->B raw_raw


,rank,score,i,j,sentence_a,sentence_b
0,1,0.807833,2,2,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
1,2,0.803327,0,0,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
2,3,0.728774,1,1,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
3,4,0.451708,1,0,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
4,5,0.432526,0,1,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...
5,6,0.424955,2,0,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,བྲག་ཕུག་གི་ཉེ་འཁོར་དུ་སྒོམ་ཆེན་ཞིག་གིས་སྔ་དྲོ་...
6,7,0.411612,0,2,རི་ཁྲོད་ཀྱི་དགོན་པ་དེར་སྔ་དྲོ་ཞིག་ལ་དགེ་སློང་ཚ...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
7,8,0.392034,1,2,ཁོང་ཚོས་བཀའ་འགྱུར་གྱི་ཚིག་དོན་ལ་ཞིབ་ཏུ་བསམས་ནས...,མཚན་མོ་ནས་ཆུ་དང་རླུང་གི་སྒྲ་མཉམ་དུ་གྲགས་པས་སེམ...
8,9,0.390660,2,1,མཚན་མོར་རླུང་གི་སྒྲ་དང་ཆུ་ཡི་སྒྲ་གཉིས་མཉམ་དུ་ཐ...,ཁོས་གསུང་རབ་ཀྱི་དོན་ལ་ཡང་ཡང་བསམས་ནས་ཏིང་ངེ་འཛི...


In [20]:
comparison = summary_df.set_index("label")[["max_score", "mean_score", "p95_score", "mean_best_row", "mean_best_col"]]
comparison

,max_score,mean_score,p95_score,mean_best_row,mean_best_col
label,,,,,
A->B query_corpus,0.7586,0.4951,0.7467,0.7234,0.7234
B->A query_corpus,0.7636,0.5047,0.7577,0.7360,0.7360
A<->B query_query,0.8143,0.5430,0.8000,0.7771,0.7771
A<->B raw_raw,0.8078,0.5382,0.8060,0.7800,0.7800


## Notes for Interpretation on this limited set

- `A -> B query_corpus` follows the current model-card retrieval usage.
- `B -> A query_corpus` tests whether reversing the query/corpus roles changes the result materially.
- `query_query` removes the role mismatch by wrapping both sides as queries.
- `raw_raw` removes the instruction wrapper and treats both sides as plain text.

If the top-ranked sentence pairs are stable across all four views, the later corpus-scale analysis is more robust. If they move around substantially, protocol choice is itself a meaningful source of uncertainty.

## Notes after the numbers are in

- On the small three-sentence control example, all four protocols recovered the same diagonal sentence alignments as the strongest matches.
- The score magnitudes moved, but the ranking structure did not. `query_query` and `raw_raw` produced the highest maxima (`0.8143` and `0.8078`), while the two `query_corpus` directions were slightly lower (`0.7586` and `0.7636`).
- That means the asymmetry is real numerically, but for this clean toy case it did not alter the substantive answer.
- This is encouraging because it suggests protocol sensitivity may be smaller than content sensitivity for clearly parallel material.

This is worth knowing: a stable ranking on a toy example does not prove that protocol choice never matters. It only justifies moving on to messier real-corpus passages.


## Real-corpus takeaways from the saved run

- Across the four sampled cross-corpus passage pairs, `A1_SMDG_rig_pa_khu_byug` vs `B2_LL01_rtse_mo_byung_rgyal` was the strongest pair overall. Its max scores were `0.5916` (`A->B query_corpus`), `0.6242` (`B->A query_corpus`), `0.6488` (`query_query`), and `0.6556` (`raw_raw`).
- `A2_SMDG_sems_lung_rgyun_thag` vs `B2_LL01_rtse_mo_byung_rgyal` was the weakest pairing across most views, with maxima between `0.4420` and `0.5453`.
- `A1 vs B1` and `A2 vs B1` sat in the middle. Their maxima and upper-tail scores suggest real conceptual overlap, but less strongly than `A1 vs B2`.
- The broad ranking of the four text pairs stayed stable across protocols even though the absolute values shifted. That makes the sample analysis look methodologically usable rather than brittle.
- `query_query` consistently produced the highest mean and max scores. `raw_raw` was usually next. The retrieval-style `query_corpus` directions were lower, but they did not reverse the high-level ordering.

Interpretively, this looks more like the model is responding to a mix of rhetorical mode and doctrinal content than to a pure borrowing signal. The strongest pair also looked strongest to a human because both passages function like elevated doctrinal openings.


In [22]:
import numpy as np
import pandas as pd
import torch

from tibetan_pipeline import TibetanResearchSDK

print("device:", sdk.device)
print("torch_dtype:", sdk.torch_dtype)
print("load_in_8bit:", sdk.load_in_8bit)
print("batch_size:", sdk.batch_size)

PASSAGES = {
    "A1_SMDG_rig_pa_khu_byug": """སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས་དང་ལྡན་པ་ལ་ཕྱག་འཚལ་ལོ། །སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས། གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི། ཡུམ་ལ་ཆགས་ནས་དངོས་པོར་བྱུང་། དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད། ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ། དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ། རིག་པ་ལྷའི་གསུང་དབྱངས་འདི། ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན། མི་གསང་གསང་བའི་གསང་མཆོག་འདི། བརྒྱ་ལམ་ན་ནི་ལན་འགའ་རུ། ཐོས་པ་དཀའ་ཡང་གོ་བར་སླ།""",
    "A2_SMDG_sems_lung_rgyun_thag": """སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ། འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན། ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་གང་ལས་དག་པ་དག སེམས་མི་འགྱུར་བའི་གཡུང་དྲུང་འདི་བསྒོམ་པར་འདོད་ན། འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས། ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས། གཤེན་ཚད་མེད་འོད་ལྡན་ལ་བསྟན་ནོ། །ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པར་བྱ། མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད། རང་ལ་འཆར་གྱི་དྲོད་མྱོང་ནས། འབྲསབུ་དག་པའི་གདོན་མི་ཟའོ།""",
    "B1_L1_rig_pa_i_khu_byug": """དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲལ་བ། གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ། མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ་ཐམས་ཅད་དངས་ཤིང་བདེན་པ་གཉིས་མགོ་མཉམ་དུ་དྲིལ་པ། ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ། ཆོས་ཐམས་ཅད་ནི་ཚུལ་བཞིན་གནས་ལ། ཆོས་ཐམས་ཅད་ཚུལ་བཞིན་དུ་མི་གནསཔའི་མཆོག །གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་""",
    "B2_LL01_rtse_mo_byung_rgyal": """རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པོ་ཀུན་ཏུ་བཟང་པོ་ལ་ཕྱག་འཚལ་ལོ། །ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །སྔོན་ཚེ་འདས་པའི་བསྐལ་པའི་མཐའ་རིང་ནས། །འཁྲུལ་འཁོར་བླུན་རྨོངས་མདོངས་པའི་འགྲོ་བ་རྣམས། །ང་དང་བདག་བཅས་མཚན་མས་བཅིངས་པ་ཡིས། །ཁམས་གསུམ་རྒྱུད་ལས་འཁོར་བའི་འགྲོ་དོན་དུ། །སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།""",
}

PAIR_CASES = [
    ("A1_SMDG_rig_pa_khu_byug", "B1_L1_rig_pa_i_khu_byug"),
    ("A1_SMDG_rig_pa_khu_byug", "B2_LL01_rtse_mo_byung_rgyal"),
    ("A2_SMDG_sems_lung_rgyun_thag", "B1_L1_rig_pa_i_khu_byug"),
    ("A2_SMDG_sems_lung_rgyun_thag", "B2_LL01_rtse_mo_byung_rgyal"),
]

def pairwise_protocol(protocol_name: str, sentences_a: list[str], sentences_b: list[str]):
    if protocol_name == "query_corpus":
        emb_a = sdk.embed_sentences(sentences_a, is_query=True)
        emb_b = sdk.embed_sentences(sentences_b, is_query=False)
    elif protocol_name == "query_query":
        emb_a = sdk.embed_sentences(sentences_a, is_query=True)
        emb_b = sdk.embed_sentences(sentences_b, is_query=True)
    elif protocol_name == "raw_raw":
        emb_a = sdk.embed_sentences(sentences_a, is_query=False)
        emb_b = sdk.embed_sentences(sentences_b, is_query=False)
    else:
        raise ValueError(protocol_name)
    return sdk.pairwise_from_embedding_views(
        emb_a,
        emb_b,
        top_k=min(12, len(sentences_a) * len(sentences_b)),
    )

def summarize_view(label: str, pair_name: str, view) -> dict[str, object]:
    matrix = view.similarity_matrix
    return {
        "pair": pair_name,
        "protocol": label,
        "shape": tuple(matrix.shape),
        "max_score": round(float(np.max(matrix)), 4) if matrix.size else None,
        "mean_score": round(float(np.mean(matrix)), 4) if matrix.size else None,
        "p95_score": round(float(np.percentile(matrix, 95)), 4) if matrix.size else None,
        "mean_best_row": round(float(np.mean(np.max(matrix, axis=1))), 4) if matrix.size else None,
        "mean_best_col": round(float(np.mean(np.max(matrix, axis=0))), 4) if matrix.size else None,
    }

def matrix_frame(view, row_label: str, col_label: str) -> pd.DataFrame:
    return pd.DataFrame(
        view.similarity_matrix,
        index=[f"{row_label}{i:02d}" for i in range(len(view.segments_a))],
        columns=[f"{col_label}{j:02d}" for j in range(len(view.segments_b))],
    )

def segment_passage(text: str) -> list[str]:
    seg = sdk.segment_text(text)
    return [s for s in seg.segments if s.strip()]

all_summaries = []
all_results = {}

for left_key, right_key in PAIR_CASES:
    left_text = PASSAGES[left_key]
    right_text = PASSAGES[right_key]
    pair_name = f"{left_key}__{right_key}"

    segments_a = segment_passage(left_text)
    segments_b = segment_passage(right_text)

    print(f"\n===== {pair_name} =====")
    print("A segments:", len(segments_a), "B segments:", len(segments_b))
    display(pd.DataFrame({"A_segments": pd.Series(segments_a), "B_segments": pd.Series(segments_b)}))

    view_ab = pairwise_protocol("query_corpus", segments_a, segments_b)
    view_ba = pairwise_protocol("query_corpus", segments_b, segments_a)
    view_qq = pairwise_protocol("query_query", segments_a, segments_b)
    view_rr = pairwise_protocol("raw_raw", segments_a, segments_b)

    pair_summary = pd.DataFrame(
        [
            summarize_view("A->B query_corpus", pair_name, view_ab),
            summarize_view("B->A query_corpus", pair_name, view_ba),
            summarize_view("A<->B query_query", pair_name, view_qq),
            summarize_view("A<->B raw_raw", pair_name, view_rr),
        ]
    )
    display(pair_summary)
    all_summaries.append(pair_summary)

    print("A->B query_corpus")
    display(matrix_frame(view_ab, "A", "B").round(4))
    print("B->A query_corpus")
    display(matrix_frame(view_ba, "B", "A").round(4))
    print("A<->B query_query")
    display(matrix_frame(view_qq, "A", "B").round(4))
    print("A<->B raw_raw")
    display(matrix_frame(view_rr, "A", "B").round(4))

    topk_tables = {
        "A->B query_corpus": view_ab.topk_dataframe(),
        "B->A query_corpus": view_ba.topk_dataframe(),
        "A<->B query_query": view_qq.topk_dataframe(),
        "A<->B raw_raw": view_rr.topk_dataframe(),
    }

    for label, table in topk_tables.items():
        print(f"## {pair_name} | {label}")
        display(table.head(10))

    all_results[pair_name] = {
        "segments_a": segments_a,
        "segments_b": segments_b,
        "view_ab": view_ab,
        "view_ba": view_ba,
        "view_qq": view_qq,
        "view_rr": view_rr,
    }

summary_df = pd.concat(all_summaries, ignore_index=True)
print("\n===== Combined Summary =====")
display(summary_df)

pivot_max = summary_df.pivot(index="pair", columns="protocol", values="max_score")
pivot_mean = summary_df.pivot(index="pair", columns="protocol", values="mean_score")

print("Max-score comparison")
display(pivot_max)

print("Mean-score comparison")
display(pivot_mean)


device: cuda
torch_dtype: float16
load_in_8bit: True
batch_size: 1

===== A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug =====
A segments: 12 B segments: 7


,A_segments,B_segments
0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
1,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
2,གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
3,ཡུམ་ལ་ཆགས་ནས་དངོས་པོར་བྱུང་།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
4,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,ཆོས་ཐམས་ཅད་ནི་ཚུལ་བཞིན་གནས་ལ།
5,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཆོས་ཐམས་ཅད་ཚུལ་བཞིན་དུ་མི་གནསཔའི་མཆོག །
6,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
7,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,NaN
8,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,NaN
9,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,NaN


[embed] backend=gemma-last-token device=cuda total_sentences=12 batch_size=1 batches=12
[embed] batch 1/12 sentences 1-1/12
[embed] batch 2/12 sentences 2-2/12
[embed] batch 3/12 sentences 3-3/12
[embed] batch 4/12 sentences 4-4/12
[embed] batch 5/12 sentences 5-5/12
[embed] batch 6/12 sentences 6-6/12
[embed] batch 7/12 sentences 7-7/12
[embed] batch 8/12 sentences 8-8/12
[embed] batch 9/12 sentences 9-9/12
[embed] batch 10/12 sentences 10-10/12
[embed] batch 11/12 sentences 11-11/12
[embed] batch 12/12 sentences 12-12/12
[embed] backend=gemma-last-token device=cuda total_sentences=7 batch_size=1 batches=7
[embed] batch 1/7 sentences 1-1/7
[embed] batch 2/7 sentences 2-2/7
[embed] batch 3/7 sentences 3-3/7
[embed] batch 4/7 sentences 4-4/7
[embed] batch 5/7 sentences 5-5/7
[embed] batch 6/7 sentences 6-6/7
[embed] batch 7/7 sentences 7-7/7
[embed] backend=gemma-last-token device=cuda total_sentences=7 batch_size=1 batches=7
[embed] batch 1/7 sentences 1-1/7
[embed] batch 2/7 sentences

,pair,protocol,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A->B query_corpus,"(12, 7)",0.4867,0.3303,0.4507,0.4158,0.4178
1,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,B->A query_corpus,"(7, 12)",0.5039,0.3313,0.4447,0.4311,0.4171
2,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A<->B query_query,"(12, 7)",0.5814,0.4068,0.5367,0.5024,0.4955
3,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A<->B raw_raw,"(12, 7)",0.5230,0.3589,0.4887,0.4479,0.4575


A->B query_corpus


,B00,B01,B02,B03,B04,B05,B06
A00,0.2263,0.2844,0.2578,0.4127,0.2135,0.2969,0.1945
A01,0.3221,0.3149,0.2667,0.4524,0.2811,0.2524,0.2698
A02,0.3561,0.3166,0.3600,0.3846,0.2654,0.2245,0.2825
A03,0.3587,0.3595,0.2538,0.3349,0.3096,0.2484,0.2947
A04,0.4313,0.3924,0.3333,0.4117,0.3629,0.3492,0.4234
A05,0.3680,0.4803,0.3233,0.4867,0.3652,0.3510,0.3040
A06,0.4277,0.4411,0.3668,0.4827,0.3657,0.3077,0.3615
A07,0.3542,0.3215,0.3618,0.4145,0.2901,0.2566,0.2844
A08,0.3750,0.3392,0.3377,0.3848,0.3274,0.2696,0.2310
A09,0.3826,0.4107,0.3467,0.4760,0.3093,0.3700,0.3317


B->A query_corpus


,A00,A01,A02,A03,A04,A05,A06,A07,A08,A09,A10,A11
B00,0.2542,0.3467,0.3589,0.3495,0.4470,0.3442,0.3906,0.3642,0.3909,0.3859,0.2790,0.3034
B01,0.3293,0.3581,0.2986,0.3686,0.3971,0.4811,0.4091,0.3225,0.3380,0.4172,0.3323,0.3350
B02,0.2799,0.3057,0.3665,0.2737,0.3645,0.3233,0.3396,0.4161,0.3688,0.3612,0.2734,0.3323
B03,0.4317,0.5039,0.3805,0.3258,0.4049,0.4568,0.4017,0.4171,0.3991,0.4647,0.3070,0.3695
B04,0.2675,0.3408,0.2693,0.3237,0.3965,0.3727,0.3527,0.3364,0.3461,0.3421,0.2734,0.2905
B05,0.3246,0.2668,0.1962,0.2325,0.3397,0.2976,0.2492,0.2723,0.2615,0.3653,0.2217,0.2363
B06,0.1984,0.2961,0.2611,0.2778,0.4079,0.2411,0.2794,0.2763,0.2237,0.3043,0.2080,0.2144


A<->B query_query


,B00,B01,B02,B03,B04,B05,B06
A00,0.2744,0.3545,0.3150,0.4823,0.2897,0.3445,0.2437
A01,0.4118,0.4226,0.3529,0.5814,0.4108,0.3433,0.3683
A02,0.4144,0.3825,0.4240,0.4759,0.3715,0.2887,0.3455
A03,0.4100,0.4199,0.3129,0.3978,0.3997,0.3017,0.3424
A04,0.4958,0.4629,0.3973,0.4971,0.4740,0.4123,0.4742
A05,0.4276,0.5480,0.3866,0.5699,0.4622,0.3943,0.3502
A06,0.4738,0.5013,0.4091,0.5403,0.4514,0.3539,0.3971
A07,0.4243,0.4142,0.4475,0.5161,0.4149,0.3459,0.3773
A08,0.4496,0.4193,0.4189,0.4947,0.4293,0.3520,0.3155
A09,0.4587,0.4883,0.4201,0.5703,0.4339,0.4472,0.4067


A<->B raw_raw


,B00,B01,B02,B03,B04,B05,B06
A00,0.2718,0.3339,0.2842,0.4382,0.2596,0.3536,0.2114
A01,0.3547,0.3595,0.3048,0.4961,0.3235,0.2802,0.2960
A02,0.3998,0.3465,0.3976,0.4047,0.2685,0.2343,0.2976
A03,0.4017,0.4198,0.3060,0.3750,0.3389,0.2863,0.3228
A04,0.4911,0.4390,0.3942,0.4388,0.4004,0.3855,0.4613
A05,0.3812,0.5230,0.3456,0.4946,0.3849,0.3629,0.2903
A06,0.4510,0.4707,0.3895,0.4749,0.3852,0.3158,0.3506
A07,0.4004,0.3547,0.4288,0.4452,0.3268,0.3000,0.2925
A08,0.4276,0.3779,0.3834,0.4153,0.3612,0.2980,0.2442
A09,0.4072,0.4544,0.3813,0.4963,0.3297,0.4014,0.3338


## A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug | A->B query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.486699,5,3,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
1,2,0.482690,6,3,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
2,3,0.480331,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
3,4,0.475966,9,3,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
4,5,0.452395,1,3,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
5,6,0.441123,6,1,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
6,7,0.431325,4,0,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
7,8,0.427658,6,0,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
8,9,0.423367,4,6,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
9,10,0.414501,7,3,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།


## A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug | B->A query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.503936,3,1,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།
1,2,0.481052,1,5,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།
2,3,0.464727,3,9,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།
3,4,0.456756,3,5,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།
4,5,0.446990,0,4,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།
5,6,0.431662,3,0,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...
6,7,0.417160,1,9,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།
7,8,0.417108,3,7,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།
8,9,0.416129,2,7,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།
9,10,0.409148,1,6,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།


## A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug | A<->B query_query


,rank,score,i,j,sentence_a,sentence_b
0,1,0.581427,1,3,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
1,2,0.570281,9,3,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
2,3,0.569854,5,3,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
3,4,0.548005,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
4,5,0.540332,6,3,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
5,6,0.516085,7,3,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
6,7,0.501349,6,1,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
7,8,0.497073,4,3,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
8,9,0.495754,4,0,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
9,10,0.494670,8,3,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།


## A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug | A<->B raw_raw


,rank,score,i,j,sentence_a,sentence_b
0,1,0.523048,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
1,2,0.496328,9,3,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
2,3,0.496071,1,3,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
3,4,0.494617,5,3,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
4,5,0.491138,4,0,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
5,6,0.474910,6,3,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
6,7,0.470657,6,1,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
7,8,0.461277,4,6,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
8,9,0.454377,9,1,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
9,10,0.450964,6,0,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...



===== A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal =====
A segments: 12 B segments: 9


,A_segments,B_segments
0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
1,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
2,གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི།,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །
3,ཡུམ་ལ་ཆགས་ནས་དངོས་པོར་བྱུང་།,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །
4,དངོས་པོ་ཉིད་ལས་རྔོས་པོ་མྱེད།,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
5,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,སྔོན་ཚེ་འདས་པའི་བསྐལ་པའི་མཐའ་རིང་ནས། །འཁྲུལ་འཁ...
6,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།,ང་དང་བདག་བཅས་མཚན་མས་བཅིངས་པ་ཡིས། །
7,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,ཁམས་གསུམ་རྒྱུད་ལས་འཁོར་བའི་འགྲོ་དོན་དུ། །
8,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,NaN


[embed] backend=gemma-last-token device=cuda total_sentences=12 batch_size=1 batches=12
[embed] batch 1/12 sentences 1-1/12
[embed] batch 2/12 sentences 2-2/12
[embed] batch 3/12 sentences 3-3/12
[embed] batch 4/12 sentences 4-4/12
[embed] batch 5/12 sentences 5-5/12
[embed] batch 6/12 sentences 6-6/12
[embed] batch 7/12 sentences 7-7/12
[embed] batch 8/12 sentences 8-8/12
[embed] batch 9/12 sentences 9-9/12
[embed] batch 10/12 sentences 10-10/12
[embed] batch 11/12 sentences 11-11/12
[embed] batch 12/12 sentences 12-12/12
[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 batches=9
[embed] batch 1/9 sentences 1-1/9
[embed] batch 2/9 sentences 2-2/9
[embed] batch 3/9 sentences 3-3/9
[embed] batch 4/9 sentences 4-4/9
[embed] batch 5/9 sentences 5-5/9
[embed] batch 6/9 sentences 6-6/9
[embed] batch 7/9 sentences 7-7/9
[embed] batch 8/9 sentences 8-8/9
[embed] batch 9/9 sentences 9-9/9
[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 bat

,pair,protocol,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A->B query_corpus,"(12, 9)",0.5916,0.3166,0.4671,0.4580,0.4589
1,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,B->A query_corpus,"(9, 12)",0.6242,0.3195,0.4598,0.4556,0.4386
2,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A<->B query_query,"(12, 9)",0.6488,0.4017,0.5609,0.5455,0.5339
3,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A<->B raw_raw,"(12, 9)",0.6556,0.3454,0.5109,0.4807,0.4922


A->B query_corpus


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.5916,0.4722,0.3786,0.3529,0.5332,0.3079,0.2612,0.2854,0.2833
A01,0.3302,0.4554,0.3073,0.2446,0.2606,0.3341,0.3150,0.3276,0.4337
A02,0.2510,0.3083,0.3181,0.2806,0.2274,0.2815,0.2920,0.3230,0.4578
A03,0.2639,0.2653,0.2308,0.2469,0.2302,0.2629,0.3459,0.3151,0.3840
A04,0.2538,0.2908,0.2777,0.2580,0.2126,0.2420,0.3649,0.2972,0.4334
A05,0.4376,0.5248,0.4894,0.3188,0.3492,0.3268,0.3025,0.3852,0.4395
A06,0.3196,0.4167,0.3636,0.4139,0.2666,0.2630,0.2923,0.3207,0.4044
A07,0.2666,0.3223,0.3045,0.3116,0.2817,0.2539,0.2836,0.2479,0.4512
A08,0.2244,0.3345,0.2576,0.2205,0.1840,0.3087,0.2852,0.3211,0.4557
A09,0.3432,0.3897,0.3439,0.3101,0.3116,0.2950,0.3316,0.3070,0.4930


B->A query_corpus


,A00,A01,A02,A03,A04,A05,A06,A07,A08,A09,A10,A11
B00,0.6242,0.3550,0.2414,0.2661,0.2505,0.4146,0.2837,0.2778,0.2303,0.3254,0.2491,0.2304
B01,0.4888,0.5245,0.3272,0.2873,0.3059,0.5019,0.3579,0.3506,0.3574,0.3777,0.2938,0.3500
B02,0.4054,0.3477,0.3339,0.2459,0.2843,0.4692,0.3148,0.3343,0.2666,0.3461,0.2488,0.2642
B03,0.3913,0.3134,0.3281,0.3120,0.3055,0.3384,0.4226,0.3707,0.2700,0.3474,0.2247,0.2509
B04,0.5402,0.2801,0.2243,0.2340,0.2195,0.3367,0.2360,0.2978,0.1982,0.2922,0.2307,0.1930
B05,0.3248,0.3511,0.2753,0.2658,0.2263,0.3008,0.2306,0.2510,0.3081,0.2998,0.3237,0.2601
B06,0.2774,0.3459,0.2730,0.3467,0.3473,0.2791,0.2393,0.2981,0.2826,0.3218,0.2343,0.2334
B07,0.3135,0.3786,0.3351,0.3174,0.2975,0.3713,0.2799,0.2800,0.3505,0.3265,0.2461,0.2827
B08,0.2766,0.4297,0.4039,0.3314,0.3888,0.3600,0.2967,0.4329,0.4113,0.4423,0.3646,0.3998


A<->B query_query


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.6488,0.5277,0.4527,0.4434,0.5956,0.3533,0.2977,0.3650,0.3380
A01,0.4066,0.5910,0.4327,0.3946,0.3592,0.4074,0.3933,0.4768,0.5542
A02,0.2979,0.4182,0.4074,0.4013,0.3167,0.3478,0.3400,0.4428,0.5340
A03,0.3001,0.3498,0.3024,0.3590,0.2930,0.3134,0.3919,0.4143,0.4539
A04,0.2965,0.3869,0.3687,0.3710,0.2803,0.2969,0.4114,0.4019,0.5128
A05,0.4882,0.6031,0.5645,0.4333,0.4254,0.3813,0.3533,0.4839,0.5145
A06,0.3631,0.4886,0.4371,0.5147,0.3312,0.3228,0.3265,0.4122,0.4725
A07,0.3417,0.4467,0.4108,0.4500,0.3831,0.3348,0.3612,0.3979,0.5511
A08,0.2819,0.4346,0.3550,0.3488,0.2803,0.3767,0.3508,0.4513,0.5431
A09,0.3962,0.4856,0.4454,0.4373,0.3840,0.3711,0.3886,0.4355,0.5757


A<->B raw_raw


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.6556,0.5286,0.4228,0.3829,0.5764,0.3373,0.3116,0.3220,0.3030
A01,0.3707,0.5177,0.3490,0.2735,0.2898,0.3568,0.3598,0.3548,0.4456
A02,0.2916,0.3401,0.3768,0.3284,0.2510,0.2887,0.3234,0.3493,0.4674
A03,0.3229,0.3181,0.2933,0.3134,0.2807,0.2894,0.4000,0.3435,0.3824
A04,0.2987,0.3296,0.3153,0.3112,0.2561,0.2494,0.4013,0.3168,0.4307
A05,0.4645,0.5521,0.5259,0.3413,0.3786,0.3241,0.3215,0.4013,0.4054
A06,0.3368,0.4247,0.3803,0.4621,0.2876,0.2557,0.3075,0.3310,0.3556
A07,0.3090,0.3674,0.3706,0.3614,0.3197,0.2531,0.3278,0.2648,0.4868
A08,0.2751,0.3900,0.2978,0.2607,0.2155,0.3217,0.3228,0.3590,0.4667
A09,0.3676,0.4117,0.3781,0.3393,0.3311,0.3023,0.3667,0.3211,0.4983


## A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal | A->B query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.591625,0,0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
1,2,0.533248,0,4,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
2,3,0.524831,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
3,4,0.493041,9,8,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
4,5,0.489443,5,2,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །
5,6,0.472150,0,1,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
6,7,0.457841,2,8,གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
7,8,0.455701,8,8,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
8,9,0.455378,1,1,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
9,10,0.451173,7,8,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།


## A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal | B->A query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.624193,0,0,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...
1,2,0.540161,4,0,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...
2,3,0.524493,1,1,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།
3,4,0.501934,1,5,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།
4,5,0.488771,1,0,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...
5,6,0.469227,2,5,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།
6,7,0.442349,8,9,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།
7,8,0.432856,8,7,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།
8,9,0.429678,8,1,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།
9,10,0.422641,3,6,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །,དབྱིངས་ཀྱི་ཐིག་ལེ་གཅིག་ལ་ཐིམ།


## A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal | A<->B query_query


,rank,score,i,j,sentence_a,sentence_b
0,1,0.648846,0,0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
1,2,0.603091,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
2,3,0.595579,0,4,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
3,4,0.590978,1,1,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
4,5,0.575693,9,8,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
5,6,0.564524,5,2,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །
6,7,0.554196,1,8,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
7,8,0.551064,7,8,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
8,9,0.543128,8,8,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,10,0.533993,2,8,གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།


## A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal | A<->B raw_raw


,rank,score,i,j,sentence_a,sentence_b
0,1,0.655591,0,0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
1,2,0.576407,0,4,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
2,3,0.552094,5,1,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
3,4,0.528634,0,1,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་ཐམས་ཅད་མཁྱེན་པའི་ཡེ་ཤེས...,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
4,5,0.525939,5,2,ཀུན་བཟང་ཐུགས་རྗེའི་སྐྱབས་འོག་ཏུ།,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །
5,6,0.517684,1,1,སྟོན་པ་ཐུགས་རྗེ་ཆེར་ལྡན་བས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
6,7,0.498347,9,8,མི་གསང་གསང་བའི་གསང་མཆོག་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
7,8,0.486802,7,8,རིག་པ་ལྷའི་གསུང་དབྱངས་འདི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
8,9,0.467366,2,8,གསུང་ལས་སྤྲུལ་པའི་ཁུ་བྱུག་ནི།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,10,0.466653,8,8,ཐབས་ཀྱིས་སྙན་ཁུངས་བརྒྱུད་པ་ཡིན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།



===== A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug =====
A segments: 9 B segments: 7


,A_segments,B_segments
0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
1,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
2,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
3,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
4,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཆོས་ཐམས་ཅད་ནི་ཚུལ་བཞིན་གནས་ལ།
5,གཤེན་ཚད་མེད་འོད་ལྡན་ལ་བསྟན་ནོ། །,ཆོས་ཐམས་ཅད་ཚུལ་བཞིན་དུ་མི་གནསཔའི་མཆོག །
6,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
7,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,NaN
8,རང་ལ་འཆར་གྱི་དྲོད་མྱོང་ནས། འབྲསབུ་དག་པའི་གདོན་...,NaN


[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 batches=9
[embed] batch 1/9 sentences 1-1/9
[embed] batch 2/9 sentences 2-2/9
[embed] batch 3/9 sentences 3-3/9
[embed] batch 4/9 sentences 4-4/9
[embed] batch 5/9 sentences 5-5/9
[embed] batch 6/9 sentences 6-6/9
[embed] batch 7/9 sentences 7-7/9
[embed] batch 8/9 sentences 8-8/9
[embed] batch 9/9 sentences 9-9/9
[embed] backend=gemma-last-token device=cuda total_sentences=7 batch_size=1 batches=7
[embed] batch 1/7 sentences 1-1/7
[embed] batch 2/7 sentences 2-2/7
[embed] batch 3/7 sentences 3-3/7
[embed] batch 4/7 sentences 4-4/7
[embed] batch 5/7 sentences 5-5/7
[embed] batch 6/7 sentences 6-6/7
[embed] batch 7/7 sentences 7-7/7
[embed] backend=gemma-last-token device=cuda total_sentences=7 batch_size=1 batches=7
[embed] batch 1/7 sentences 1-1/7
[embed] batch 2/7 sentences 2-2/7
[embed] batch 3/7 sentences 3-3/7
[embed] batch 4/7 sentences 4-4/7
[embed] batch 5/7 sentences 5-5/7
[embed] batch 6/7 sentences 

,pair,protocol,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,A->B query_corpus,"(9, 7)",0.4882,0.3290,0.4407,0.4121,0.4306
1,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,B->A query_corpus,"(7, 9)",0.4874,0.3444,0.4427,0.4316,0.4267
2,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,A<->B query_query,"(9, 7)",0.5594,0.4040,0.5070,0.4855,0.5014
3,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,A<->B raw_raw,"(9, 7)",0.5211,0.3601,0.4766,0.4474,0.4665


A->B query_corpus


,B00,B01,B02,B03,B04,B05,B06
A00,0.3453,0.3065,0.4103,0.4706,0.3016,0.2816,0.3546
A01,0.3498,0.3102,0.3336,0.3588,0.2594,0.2199,0.3123
A02,0.3553,0.3710,0.3683,0.4166,0.3268,0.3179,0.3891
A03,0.2822,0.2707,0.2443,0.3855,0.2843,0.2070,0.2591
A04,0.3259,0.3516,0.2514,0.4029,0.2559,0.2308,0.2689
A05,0.3659,0.3177,0.2595,0.3260,0.2173,0.3045,0.2857
A06,0.3318,0.2933,0.4537,0.3458,0.3447,0.3047,0.2765
A07,0.4882,0.4265,0.4418,0.3517,0.3625,0.3816,0.4309
A08,0.3671,0.2981,0.3401,0.3638,0.2847,0.2594,0.3213


B->A query_corpus


,A00,A01,A02,A03,A04,A05,A06,A07,A08
B00,0.3537,0.3491,0.3645,0.3253,0.3201,0.4005,0.3339,0.4647,0.3812
B01,0.3271,0.3042,0.3964,0.3229,0.3661,0.3508,0.2989,0.4237,0.3068
B02,0.4284,0.3441,0.3584,0.2887,0.2601,0.2986,0.4640,0.4280,0.3468
B03,0.4874,0.3472,0.4367,0.4434,0.4130,0.3575,0.3475,0.3146,0.3611
B04,0.3639,0.3033,0.3893,0.3729,0.2885,0.2912,0.3650,0.3992,0.3324
B05,0.2889,0.2333,0.3362,0.2692,0.1975,0.3428,0.2777,0.3749,0.2428
B06,0.3533,0.2925,0.4071,0.2972,0.2865,0.2963,0.2835,0.3947,0.3051


A<->B query_query


,B00,B01,B02,B03,B04,B05,B06
A00,0.4018,0.3805,0.4761,0.5594,0.4171,0.3407,0.4151
A01,0.4135,0.3809,0.3927,0.4306,0.3728,0.2965,0.3775
A02,0.4008,0.4196,0.4078,0.4699,0.4169,0.3585,0.4326
A03,0.3772,0.3808,0.3420,0.5072,0.4249,0.3208,0.3811
A04,0.3960,0.4383,0.3235,0.5054,0.3704,0.3039,0.3560
A05,0.4318,0.4006,0.3348,0.4320,0.3403,0.3877,0.3645
A06,0.3963,0.3634,0.5071,0.4258,0.4329,0.3546,0.3403
A07,0.5416,0.5002,0.4908,0.4300,0.4700,0.4512,0.4801
A08,0.4153,0.3437,0.3734,0.4167,0.3781,0.3008,0.3584


A<->B raw_raw


,B00,B01,B02,B03,B04,B05,B06
A00,0.3805,0.3451,0.4384,0.5002,0.3378,0.3194,0.3770
A01,0.3782,0.3364,0.3668,0.3736,0.2849,0.2543,0.3153
A02,0.3709,0.4024,0.3694,0.4374,0.3509,0.3479,0.4097
A03,0.3293,0.3295,0.2810,0.4308,0.3419,0.2634,0.2806
A04,0.3451,0.3864,0.2761,0.4302,0.2810,0.2267,0.2972
A05,0.4262,0.3699,0.3060,0.3551,0.2574,0.3605,0.3042
A06,0.3520,0.3183,0.4916,0.3652,0.3657,0.3157,0.2988
A07,0.5211,0.4705,0.4772,0.3610,0.4054,0.4226,0.4538
A08,0.4107,0.3479,0.3860,0.3983,0.3216,0.2838,0.3470


## A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug | A->B query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.488233,7,0,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
1,2,0.470608,0,3,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
2,3,0.453726,6,2,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
3,4,0.441808,7,2,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
4,5,0.430860,7,6,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
5,6,0.426547,7,1,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
6,7,0.416626,2,3,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
7,8,0.410251,0,2,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
8,9,0.402854,4,3,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
9,10,0.389051,2,6,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་


## A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug | B->A query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.487422,3,0,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།
1,2,0.464747,0,7,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།
2,3,0.463962,2,6,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...
3,4,0.443398,3,3,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།
4,5,0.436720,3,2,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...
5,6,0.428368,2,0,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།
6,7,0.427962,2,7,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།
7,8,0.423702,1,7,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།
8,9,0.413012,3,4,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།
9,10,0.407051,6,2,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...


## A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug | A<->B query_query


,rank,score,i,j,sentence_a,sentence_b
0,1,0.559378,0,3,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
1,2,0.541616,7,0,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
2,3,0.507219,3,3,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
3,4,0.507147,6,2,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
4,5,0.505371,4,3,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
5,6,0.500237,7,1,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
6,7,0.490850,7,2,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
7,8,0.480136,7,6,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
8,9,0.476087,0,2,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
9,10,0.469957,7,4,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,ཆོས་ཐམས་ཅད་ནི་ཚུལ་བཞིན་གནས་ལ།


## A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug | A<->B raw_raw


,rank,score,i,j,sentence_a,sentence_b
0,1,0.521142,7,0,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དོག་དང་དབྱིབས་དང་མཚན་མ་མེད་པར་རིགས་རྒྱུད་དང་བྲ...
1,2,0.500186,0,3,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
2,3,0.491575,6,2,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
3,4,0.477222,7,2,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
4,5,0.470519,7,1,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་གི་གནས་སུ་མ་གྱུར་ཅིང་ཐམས་ཅད་ཀྱི་གཞིར་གྱུར་པ།
5,6,0.453765,7,6,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,གང་བསམ་དུ་མེད་པའི་ངོ་བོ་ཉིད་
6,7,0.438399,0,2,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,མིང་དང་ཚིག་གིས་རྣམ་པར་དག་ཅིང་། རྣམ་པའི་སྤྱོད་པ...
7,8,0.437380,2,3,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
8,9,0.430760,3,3,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།
9,10,0.430184,4,3,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཡེ་ཤེས་སྙིང་པོ་ཆེ་བ་ཐམས་ཅད་ཀྱི་བླ།



===== A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal =====
A segments: 9 B segments: 9


,A_segments,B_segments
0,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
1,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
2,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །
3,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །
4,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
5,གཤེན་ཚད་མེད་འོད་ལྡན་ལ་བསྟན་ནོ། །,སྔོན་ཚེ་འདས་པའི་བསྐལ་པའི་མཐའ་རིང་ནས། །འཁྲུལ་འཁ...
6,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,ང་དང་བདག་བཅས་མཚན་མས་བཅིངས་པ་ཡིས། །
7,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,ཁམས་གསུམ་རྒྱུད་ལས་འཁོར་བའི་འགྲོ་དོན་དུ། །
8,རང་ལ་འཆར་གྱི་དྲོད་མྱོང་ནས། འབྲསབུ་དག་པའི་གདོན་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།


[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 batches=9
[embed] batch 1/9 sentences 1-1/9
[embed] batch 2/9 sentences 2-2/9
[embed] batch 3/9 sentences 3-3/9
[embed] batch 4/9 sentences 4-4/9
[embed] batch 5/9 sentences 5-5/9
[embed] batch 6/9 sentences 6-6/9
[embed] batch 7/9 sentences 7-7/9
[embed] batch 8/9 sentences 8-8/9
[embed] batch 9/9 sentences 9-9/9
[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 batches=9
[embed] batch 1/9 sentences 1-1/9
[embed] batch 2/9 sentences 2-2/9
[embed] batch 3/9 sentences 3-3/9
[embed] batch 4/9 sentences 4-4/9
[embed] batch 5/9 sentences 5-5/9
[embed] batch 6/9 sentences 6-6/9
[embed] batch 7/9 sentences 7-7/9
[embed] batch 8/9 sentences 8-8/9
[embed] batch 9/9 sentences 9-9/9
[embed] backend=gemma-last-token device=cuda total_sentences=9 batch_size=1 batches=9
[embed] batch 1/9 sentences 1-1/9
[embed] batch 2/9 sentences 2-2/9
[embed] batch 3/9 sentences 3-3/9
[embed] batch 4/9 sentences 

,pair,protocol,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_...,A->B query_corpus,"(9, 9)",0.4853,0.3135,0.4257,0.4275,0.4026
1,A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_...,B->A query_corpus,"(9, 9)",0.4420,0.3270,0.4288,0.4141,0.4149
2,A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_...,A<->B query_query,"(9, 9)",0.5453,0.3927,0.5044,0.5022,0.4712
3,A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_...,A<->B raw_raw,"(9, 9)",0.5015,0.3435,0.4534,0.4480,0.4315


A->B query_corpus


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.3285,0.4024,0.3530,0.2457,0.3994,0.2941,0.3132,0.2779,0.4400
A01,0.2476,0.2906,0.2218,0.2145,0.2670,0.2170,0.2855,0.2990,0.4853
A02,0.3292,0.3915,0.3931,0.3169,0.4359,0.3817,0.3103,0.3560,0.4535
A03,0.2145,0.2974,0.2293,0.2184,0.2363,0.2517,0.3014,0.2226,0.4086
A04,0.3144,0.4139,0.2880,0.2482,0.2840,0.3115,0.3538,0.3071,0.3911
A05,0.3842,0.3716,0.3296,0.3500,0.3295,0.3236,0.3079,0.3046,0.3472
A06,0.2360,0.2799,0.3193,0.2707,0.2964,0.2551,0.2956,0.3266,0.4257
A07,0.2844,0.3096,0.3168,0.4191,0.2606,0.2444,0.3203,0.2721,0.3539
A08,0.2691,0.3145,0.2568,0.2281,0.3179,0.2502,0.3136,0.2515,0.4176


B->A query_corpus


,A00,A01,A02,A03,A04,A05,A06,A07,A08
B00,0.3391,0.2563,0.3468,0.2748,0.2983,0.4070,0.2442,0.2784,0.2569
B01,0.4288,0.3033,0.4262,0.3677,0.4420,0.3989,0.2739,0.3066,0.3254
B02,0.3726,0.2305,0.4231,0.3170,0.3067,0.3446,0.3168,0.3074,0.2511
B03,0.3027,0.2605,0.3679,0.3302,0.2908,0.3943,0.3106,0.4178,0.2597
B04,0.4049,0.2917,0.4400,0.2872,0.2829,0.3592,0.3209,0.2670,0.3212
B05,0.3150,0.2244,0.4037,0.2803,0.2943,0.3489,0.2495,0.2448,0.2493
B06,0.3100,0.2776,0.3020,0.3304,0.3634,0.3229,0.2833,0.3116,0.3194
B07,0.3156,0.3073,0.3910,0.3010,0.3287,0.3455,0.3448,0.2793,0.2659
B08,0.4156,0.4385,0.4388,0.3990,0.3482,0.3303,0.3801,0.2922,0.3811


A<->B query_query


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.3867,0.4957,0.4382,0.3644,0.4765,0.3493,0.3519,0.3798,0.5044
A01,0.2948,0.3762,0.3031,0.3210,0.3456,0.2715,0.3323,0.4018,0.5453
A02,0.3666,0.4575,0.4529,0.4096,0.4889,0.4295,0.3349,0.4279,0.4972
A03,0.3016,0.4332,0.3702,0.3790,0.3504,0.3362,0.3872,0.3858,0.5247
A04,0.3754,0.5137,0.3843,0.3751,0.3707,0.3670,0.4186,0.4282,0.4875
A05,0.4500,0.4644,0.4258,0.4661,0.4180,0.3827,0.3640,0.4221,0.4258
A06,0.2993,0.3591,0.4011,0.3815,0.3842,0.3066,0.3486,0.4249,0.4871
A07,0.3429,0.4137,0.4066,0.5139,0.3473,0.3080,0.3701,0.3899,0.4355
A08,0.2917,0.3713,0.3082,0.3070,0.3714,0.2895,0.3400,0.3311,0.4678


A<->B raw_raw


,B00,B01,B02,B03,B04,B05,B06,B07,B08
A00,0.3619,0.4410,0.3883,0.2770,0.4196,0.3250,0.3519,0.3158,0.4628
A01,0.2864,0.3247,0.2534,0.2510,0.3088,0.2347,0.3179,0.3151,0.5015
A02,0.3566,0.4148,0.4194,0.3287,0.4431,0.3948,0.3243,0.3804,0.4534
A03,0.2834,0.3587,0.3023,0.2727,0.2855,0.2697,0.3510,0.2649,0.4140
A04,0.3298,0.4595,0.3306,0.2757,0.3015,0.3130,0.3925,0.3298,0.3793
A05,0.4383,0.4201,0.3678,0.3912,0.3875,0.3646,0.3607,0.3511,0.3659
A06,0.2602,0.2954,0.3364,0.2928,0.3260,0.2639,0.3146,0.3535,0.4218
A07,0.3166,0.3335,0.3496,0.4537,0.2993,0.2656,0.3679,0.2955,0.3407
A08,0.3103,0.3598,0.2930,0.2611,0.3554,0.2689,0.3676,0.2817,0.4271


## A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal | A->B query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.485317,1,8,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
1,2,0.453463,2,8,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
2,3,0.439961,0,8,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
3,4,0.435896,2,4,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
4,5,0.425679,6,8,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
5,6,0.419089,7,3,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །
6,7,0.417624,8,8,རང་ལ་འཆར་གྱི་དྲོད་མྱོང་ནས། འབྲསབུ་དག་པའི་གདོན་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
7,8,0.413929,4,1,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
8,9,0.408619,3,8,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,10,0.402398,0,1,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །


## A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal | B->A query_corpus


,rank,score,i,j,sentence_a,sentence_b
0,1,0.442000,1,4,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།
1,2,0.439962,4,2,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...
2,3,0.438802,8,2,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...
3,4,0.438472,8,1,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།
4,5,0.428823,1,0,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།
5,6,0.426187,1,2,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...
6,7,0.423091,2,2,སྣ་ཚོགས་ཀུན་ཏུ་བཟང་པོ་རྡོ་རྗེ་སེམས། །,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...
7,8,0.417789,3,7,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།
8,9,0.415616,8,0,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།
9,10,0.406953,0,5,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...,གཤེན་ཚད་མེད་འོད་ལྡན་ལ་བསྟན་ནོ། །


## A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal | A<->B query_query


,rank,score,i,j,sentence_a,sentence_b
0,1,0.545302,1,8,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
1,2,0.524725,3,8,འདི་ལྟར་ཤེས་པར་བྱའོ་ཞེས།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
2,3,0.513916,7,3,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །
3,4,0.513673,4,1,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
4,5,0.504412,0,8,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
5,6,0.497158,2,8,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
6,7,0.495697,0,1,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
7,8,0.488886,2,4,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
8,9,0.487473,4,8,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,10,0.487148,6,8,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།


## A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal | A<->B raw_raw


,rank,score,i,j,sentence_a,sentence_b
0,1,0.501510,1,8,འབྲས་བུ་ངེས་པར་སྟོན་པའི་དོན།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
1,2,0.462783,0,8,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
2,3,0.459517,4,1,ཐུགས་རྗེས་བྱིན་གྱིས་བརླབས་ཀྱིས།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
3,4,0.453678,7,3,མ་གཡོས་མ་བཅོས་གཉིས་སུ་མྱེད།,དབྱིངས་དང་ཡེ་ཤེས་འདུ་འབྲལ་མེད་པའི་སྐུ། །
4,5,0.453428,2,8,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
5,6,0.443139,2,4,ཕྱིས་བྱོན་པའི་ལས་དག་པ་རྣམས་ལ་སྟོན་ཏེ། ཨེ་མ་ཧོ་...,རྣམ་དག་བྱང་ཆུབ་སེམས་ལ་ཕྱག་འཚལ་ལོ། །
6,7,0.440994,0,1,སྟོན་པ་གཤེན་ལྷ་འོད་དཀར་གྱི་ཐུགས་ཡང་དག་པའི་བརྣག་པ།,ཐམས་ཅད་མཁྱེན་པ་བདེ་བ་ཆེན་པོའི་ཐུགས། །
7,8,0.438267,5,0,གཤེན་ཚད་མེད་འོད་ལྡན་ལ་བསྟན་ནོ། །,རྩེ་མོ་བྱུང་རྒྱལ། བཅོམ་ལྡན་འདས་དཔལ་བདེ་བ་ཆེན་པ...
8,9,0.427078,8,8,རང་ལ་འཆར་གྱི་དྲོད་མྱོང་ནས། འབྲསབུ་དག་པའི་གདོན་...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།
9,10,0.421822,6,8,ལྟ་སྤྱོད་དམ་ཚིག་འཕྲིན་ལས་དག སྒོམ་པ་དག་ཏུ་ཤེས་པ...,སྙིགས་མའི་དོན་ལ་གདམས་པའི་གསུང་བརྗོད་འདི།



===== Combined Summary =====


,pair,protocol,shape,max_score,mean_score,p95_score,mean_best_row,mean_best_col
0,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A->B query_corpus,"(12, 7)",0.4867,0.3303,0.4507,0.4158,0.4178
1,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,B->A query_corpus,"(7, 12)",0.5039,0.3313,0.4447,0.4311,0.4171
2,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A<->B query_query,"(12, 7)",0.5814,0.4068,0.5367,0.5024,0.4955
3,A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,A<->B raw_raw,"(12, 7)",0.5230,0.3589,0.4887,0.4479,0.4575
4,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A->B query_corpus,"(12, 9)",0.5916,0.3166,0.4671,0.4580,0.4589
5,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,B->A query_corpus,"(9, 12)",0.6242,0.3195,0.4598,0.4556,0.4386
6,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A<->B query_query,"(12, 9)",0.6488,0.4017,0.5609,0.5455,0.5339
7,A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung...,A<->B raw_raw,"(12, 9)",0.6556,0.3454,0.5109,0.4807,0.4922
8,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,A->B query_corpus,"(9, 7)",0.4882,0.3290,0.4407,0.4121,0.4306
9,A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_k...,B->A query_corpus,"(7, 9)",0.4874,0.3444,0.4427,0.4316,0.4267


Max-score comparison


protocol,A->B query_corpus,A<->B query_query,A<->B raw_raw,B->A query_corpus
pair,,,,
A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,0.4867,0.5814,0.5230,0.5039
A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal,0.5916,0.6488,0.6556,0.6242
A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug,0.4882,0.5594,0.5211,0.4874
A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal,0.4853,0.5453,0.5015,0.4420


Mean-score comparison


protocol,A->B query_corpus,A<->B query_query,A<->B raw_raw,B->A query_corpus
pair,,,,
A1_SMDG_rig_pa_khu_byug__B1_L1_rig_pa_i_khu_byug,0.3303,0.4068,0.3589,0.3313
A1_SMDG_rig_pa_khu_byug__B2_LL01_rtse_mo_byung_rgyal,0.3166,0.4017,0.3454,0.3195
A2_SMDG_sems_lung_rgyun_thag__B1_L1_rig_pa_i_khu_byug,0.3290,0.4040,0.3601,0.3444
A2_SMDG_sems_lung_rgyun_thag__B2_LL01_rtse_mo_byung_rgyal,0.3135,0.3927,0.3435,0.3270


## What the numbers justify doing next

- The notebook supports using the SDK as the inner engine for larger corpus analysis.
- It does **not** justify collapsing the corpus question to one similarity score or treating top-k matches as direct proof of copying.
- The next runner should save one sentence-embedding artifact per file, one `N x M` matrix per text pair, and a summary table with at least `max`, `p95`, `mean_best_row`, `mean_best_col`, and a top-k aggregate.
- At corpus scale, the main statistical question becomes whether the strongest cross-corpus pairs are exceptional relative to the background distribution, not whether any one raw score merely looks high.
- The most valuable human follow-up is to inspect whether the top-ranked sentence pairs are generic doctrinal-opening language, or whether they contain more specific and historically informative conceptual or phrase-level overlap.
